File: lib/core/constants/economy_constants.dart
abstract final class EconomyConstants {
  static const signupRewardSrv = 100;
  static const trialRunsRequired = 5;
  static const trialCompletionRewardSrv = 500;
  static const referralRewardSrv = 300;
  static const maxReferralPayouts = 10;

  /// 마이페이지 닉네임 변경 수수료 (DIA).
  static const nicknameChangeFeeDia = 100;

  static const dailyCapKm = 5.0;
  static const dailyCapSrvTokens = 50;
  static const srvTokensPerKm = 10;

  /// 만보기 채굴 — 0.1km당 1 SHARE, 보폭 추정.
  static const pedometerCoinIntervalKm = 0.1;
  static const pedometerSharePerCoin = 1;
  static const pedometerStrideMeters = 0.75;

  /// 혼자 뛰기/걷기 출석 인정 최소 거리.
  static const streakMinKm = 1.0;
  static const streakBonusDays = 7;
  static const streakBonusDia = 10;

  /// 러닝화 수명 경보 — 500km의 80%.
  static const shoeMileageWarnKm = 400.0;
  static const defaultPreferredRunHour = 19;

  static const diamondRefundPolicy =
      '구매 후 7일 이내, 100% 미사용 분에 한해 환불 가능';
}

File: backend/jena_ai/app/constants/economy_constants.py
"""Onboarding rewards, referral limits, and daily SRV mining cap."""

SIGNUP_REWARD_SRV = 100
TRIAL_RUNS_REQUIRED = 5
TRIAL_COMPLETION_REWARD_SRV = 500
REFERRAL_REWARD_SRV = 300
MAX_REFERRAL_PAYOUTS = 10

DAILY_CAP_KM = 5.0
DAILY_CAP_SRV_TOKENS = 50
SRV_TOKENS_PER_KM = 10

# Walking challenge harvest: 0.1 SHARE per 10 steps (1 SHARE / 100 steps).
PEDOMETER_SHARE_PER_STEP = 0.01
PEDOMETER_DAILY_HARVEST_SHARE_CAP = 60

# Debug one-shot QA grant. Play Store / release clients never send the
# baked debug-client secret. UID allowlist remains an optional extra gate.
TEST_WALLET_GRANT_AMOUNT = 1_000_000
TEST_WALLET_GRANT_FLAG = "testGrant1mDone"
TEST_WALLET_GRANT_ELIGIBLE_FLAG = "testGrant1mEligible"
TEST_WALLET_GRANT_UIDS: frozenset[str] = frozenset()
# Must match Flutter DebugTestWalletGrantHost.debugClientSecret (kDebugMode).
TEST_WALLET_GRANT_DEBUG_CLIENT_SECRET = "sharerun-debug-test-grant-1m"
